lms/djangoapps/teams/views.py

```diff
@@ -168,9 +168,10 @@ def get(self, request, course_id):
             "countries": list(countries),
             "disable_courseware_js": True,
             "teams_base_url": reverse('teams_dashboard', request=request, kwargs={'course_id': course_id}),
-            "rocket_chat_locator": ModifyTeams(request, user, course_key).get_rocket_chat_locator(),
+            "replacement_component_locator": get_replacement_location_id(course_key),
```

Comment:
here to get replacement_component_locator you use the method get_replacement_location_id, but to get disable_default_discussion in the code below, you do all the inline operations to get the setting (get the course, the teams configuration and finally the required key). what do you think about centralize the way all these settings are extracted from course in the team_features.py file? there what we can do is to fetch the complete teams_configuration object and return the required teams_configuration key

Reply:
@jfavellar90, Maybe there is a problem with the method name because get_replacement_location_id does not return the location id, this method returns a BlockUsage Locator and it's very different to the location id that's why I didn't use the inline operation.